### ISSUE TITLE
Bug parsing command line when docker daemon running with --log-driver=journald

### ISSUE THREAD

**Issue by mkocikowski** [docker/docker#22487]
`--log-opt tag=foo` throws error, saying that driver is json-file, even though it is journald. 

**Output of `docker version`:**

```
Client:
 Version:      1.11.1
 API version:  1.23
 Go version:   go1.5.4
 Git commit:   5604cbe
 Built:        Tue Apr 26 23:38:55 2016
 OS/Arch:      linux/amd64

Server:
 Version:      1.11.1
 API version:  1.23
 Go version:   go1.5.4
 Git commit:   5604cbe
 Built:        Tue Apr 26 23:38:55 2016
 OS/Arch:      linux/amd64
```

**Output of `docker info`:**

```
Containers: 7
 Running: 5
 Paused: 0
 Stopped: 2
Images: 9
Server Version: 1.11.1
Storage Driver: devicemapper
 Pool Name: docker-8:1-493088-pool
 Pool Blocksize: 65.54 kB
 Base Device Size: 10.74 GB
 Backing Filesystem: ext4
 Data file: /dev/loop0
 Metadata file: /dev/loop1
 Data Space Used: 1.356 GB
 Data Space Total: 107.4 GB
 Data Space Available: 38.45 GB
 Metadata Space Used: 3.375 MB
 Metadata Space Total: 2.147 GB
 Metadata Space Available: 2.144 GB
 Udev Sync Supported: true
 Deferred Removal Enabled: false
 Deferred Deletion Enabled: false
 Deferred Deleted Device Count: 0
 Data loop file: /var/lib/docker/devicemapper/devicemapper/data
 WARNING: Usage of loopback devices is strongly discouraged for production use. Either use `--storage-opt dm.thinpooldev` or use `--storage-opt dm.no_warn_on_loop_devices=true` to suppress this warning.
 Metadata loop file: /var/lib/docker/devicemapper/devicemapper/metadata
 Library Version: 1.02.99 (2015-06-20)
Logging Driver: journald
Cgroup Driver: cgroupfs
Plugins: 
 Volume: local
 Network: bridge null host weavemesh
Kernel Version: 4.2.0-30-generic
Operating System: Ubuntu 15.10
OSType: linux
Architecture: x86_64
CPUs: 1
Total Memory: 1.954 GiB
Name: datapipe
ID: E2GL:IOBA:6VIK:C4F5:R42Q:LLA5:OSWE:TBSO:KXUL:N2PB:5MCI:ATUN
Docker Root Dir: /var/lib/docker
Debug mode (client): false
Debug mode (server): false
Registry: https://index.docker.io/v1/
WARNING: No swap limit support
```

**Additional environment details (AWS, VirtualBox, physical, etc.):**
Vagrant --> VirtualBox

**Steps to reproduce the issue:**
1. run docker with journald driver: `/usr/bin/docker daemon -H fd:// --log-driver journald`
2. try to launch container:
```
docker run -it --rm --log-opt tag=foo alpine sh
```

**Describe the results you received:**
```
docker: Error response from daemon: unknown log opt 'tag' for json-file log driver.
See 'docker run --help'.
```

**Describe the results you expected:**
launch container with no problem - log-driver is inherited from docker daemon. when container is launched without `--log-opt`, `docker inspect` shows the driver as journald

**Additional information you deem important (e.g. issue happens only occasionally):**

**Comment by thaJeztah** [comment]
Thanks for reporting; this looks like a duplicate of https://github.com/docker/docker/issues/21118, and should be fixed by https://github.com/docker/docker/pull/21153, which will be in docker 1.12

**Comment by thaJeztah** [comment]
I'll close this issue, because this should be resolved, but feel free to comment after I closed

**Comment by mkocikowski** [comment]
you are right. thank you!

On Tue, May 3, 2016 at 4:01 PM, Sebastiaan van Stijn <
notifications@github.com> wrote:

> Closed #22487 <https://github.com/docker/docker/issues/22487>.
>
> —
> You are receiving this because you authored the thread.
> Reply to this email directly or view it on GitHub
> <https://github.com/docker/docker/issues/22487#event-649533796>
>



-- 
GPG Pub @ pgp.mit.edu F/P:   8CD2 90A8 C88E 51EB DEB6  4905 34B9 2BC3 CC28
A586
